project/api/models.py

```diff
@@ -196,3 +173,66 @@ def to_json(self):
     def save_to_db(self):
         db.session.add(self)
         db.session.commit()
+
+
+class WorksModel(db.Model):
+    __tablename__ = 'works'
+
+    provider_category_id = db.Column(db.Integer, db.ForeignKey(
+        'PROVIDER_CATEGORY.provider_category_id'), nullable=False, primary_key=True)
+    provider_id = db.Column(db.Integer, db.ForeignKey(
+        'PROVIDER.provider_id'), nullable=False, primary_key=True)
+
+    def __init__(self, provider_category_id, provider_id):
+        self.provider_category_id = provider_category_id
+        self.provider_id = provider_id
+
+    def to_json(self):
+        return{
+            'provider_category_id': self.provider_category_id,
+            'provider_id': self.provider_id
+        }
+
+    def save_to_db(self):
+        db.session.add(self)
+        db.session.commit()
+
+
+class Category():
+
+    def to_json(self):
+        pass
+
+
+class ProviderCategory(db.Model, Category):
+    __tablename__ = 'PROVIDER_CATEGORY'
+
+    provider_category_id = db.Column(
+        db.Integer, primary_key=True, autoincrement=True)
+    name = db.Column(db.String(50), nullable=False)
+    general_category_id = db.Column(
+        db.Integer, db.ForeignKey('general_category.id'))
+
+    def to_json(self, category):
+
+        return {
+            'generalId': category.general_category_id,
+            'id': category.provider_category_id,
+            'name': category.name,
+        }
+
+
+class GeneralCategory(db.Model, Category):
+
+    __tablename__ = 'GENERAL_CATEGORY'
+
+    general_category_id = db.Column(
+        db.Integer, primary_key=True, autoincrement=True)
+    name = db.Column(db.String(50), nullable=False)
+
+    def to_json(self, category):
+
+        return {
+            'id': category.general_category_id,
+            'name': category.name,
+        }
```

Comment:
Mano, essa responsabilidade não é do serviço de User, mas sim do de Categorias! O Fepas e a Fabi já trabalharam nas [models](https://github.com/pax-app/Category/blob/devel/project/api/models.py). Tenta remodelar sua lógica pra poder usar o que for necessário de cada serviço, podendo usar a Gateway também. Acho melhor você ir pelo Front primeiro, porque lá você vai ter acesso ao provider_id e provider_category_id, se já tiver listados as categorias.

Reply:
Então roger, pelo que percebi pra usar o `db.ForeignKey` é necessário ter a model definida no projeto, por isso tive que definir as tabelas

Se eu utilizasse o serviço do category pra guardar as categorias, eu teria que definir a model do provider lá do mesmo jeito, ai pra evitar realizar 2 requisições no front eu decidi trazer as models do category pra cá, sei que fica com repetição de código mas não encontrei outro jeito de fazer.

Reply:
@gabrielalbino concordo com o @rogerioo , entendo que você precise dessa comunicação com o serviço de categorias, mas não é responsabilidade do serviço de usuários armazená-las de fato. Você pode fazer uma lógica semelhante a que [eu fiz](https://github.com/pax-app/User/blob/devel/project/api/utils/display_strategy.py#L31) para ordenação dos prestadores de serviço por categoria, que necessita de uma requisição para o serviço de [review](https://github.com/pax-app/Review). Podemos discutir hoje a noite uma solução para o seu problema por hangouts, pois previamente eu estava pensando exatamente nisso.

Reply:
Uma solução para esse problema poderia ser resolvido mediante o uso da Gateway! Pois acredito que no User deveria ter somente os relacionamentos da tabela Works e as chaves continuam definidas no Category.